extensions/markdown-language-features/src/markdownEngine.ts

```diff
@@ -251,7 +251,7 @@ export class MarkdownEngine {
 						}
 					}
 
-					if (uri.fragment) {
+					if (uri.fragment && !path.extname(uri.fsPath)) {
```

Comment:
We should also normalize links that appear to be markdown files from their extension (such as ending in `.md`)

Reply:
Should we do this for all the possible markdown extensions (as recognized by vscode), or just for `.md`?

https://github.com/microsoft/vscode/blob/5c9a1b5d40f06c92c66bb2dc5f80c92eac374b88/extensions/markdown-basics/package.json#L19-L29

Reply:
Sure, seems like a good idea